### PR TITLE
Increase code covergae

### PR DESCRIPTION
**Pull request by anshumansworld**
This includes addition of more tests for Banner, FeedBanner & Interstitial in order to increase the code coverage.

### CODE REVIEW

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java [hunk 1]
```diff
@@ -39,12 +41,16 @@
...
 import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 @RunWith(RobolectricGradleTestRunner.class)
 @Config(constants = BuildConfig.class,
         sdk = 21)
 public class PubnativeNetworkFeedBannerTest {
 
+    final static String TEST_PLACEMENT_ID_INVALID = "mPlacementName";
+    final static String TEST_APP_TOKEN            = "app_token";
```
Comment (RagnarR) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java]: I think better use the same style as in other tests - not use constant string. Put it in each test as is. @cerberillo am I right?

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkInterstitialTest.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+@RunWith(RobolectricGradleTestRunner.class)
+@Config(constants = BuildConfig.class,
+        sdk = 21)
+public class PubnativeNetworkInterstitialTest {
+
+    final static String TEST_PLACEMENT_ID_INVALID = "mPlacementName";
+    final static String TEST_APP_TOKEN            = "app_token";
```
Comment (RagnarR) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkInterstitialTest.java]: The same situation as above.

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkBannerTest.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+        PubnativeNetworkBanner request = spy(PubnativeNetworkBanner.class);
+        PubnativeNetworkBanner.Listener listener = spy(PubnativeNetworkBanner.Listener.class);
+        request.mHandler = new Handler();
+        request.mListener = listener;
+
+        request.load(null, TEST_APP_TOKEN, TEST_PLACEMENT_ID_INVALID);
+
+        verify(listener).onPubnativeNetworkBannerLoadFail(eq(request), eq(PubnativeException.BANNER_PARAMETERS_INVALID));
+    }
```
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkBannerTest.java]: you could also complete this by adding 4 more tests, for empty and null app token and placement

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkBannerTest.java [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+        PubnativeNetworkBanner request = spy(PubnativeNetworkBanner.class);
+        PubnativeNetworkBanner.Listener listener = spy(PubnativeNetworkBanner.Listener.class);
+        request.mHandler = new Handler();
+        request.mListener = listener;
+
+        request.load(RuntimeEnvironment.application.getApplicationContext(), "", "");
+
+        verify(listener).onPubnativeNetworkBannerLoadFail(eq(request), eq(PubnativeException.BANNER_PARAMETERS_INVALID));
+    }
```
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkBannerTest.java]: This are 2 different tests, all valid except 1, otherwise you're just testing when explicitly 2 are invalid

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java [hunk 5]
```diff
@@ -150,4 +156,52 @@ public void invokeClick_withValidListener_callbackClick() {
...
+
+    @Test
+    public void load_withInvalidParams_callBackLoadFail() {
+
+        PubnativeNetworkFeedBanner request = spy(PubnativeNetworkFeedBanner.class);
+        PubnativeNetworkFeedBanner.Listener listener = spy(PubnativeNetworkFeedBanner.Listener.class);
+        request.mHandler = new Handler();
+        request.mListener = listener;
+
+        request.load(RuntimeEnvironment.application.getApplicationContext(), "", "");
```
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java]: Same here, this is 2 different tests

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java [hunk 6]
```diff
@@ -150,4 +156,52 @@ public void invokeClick_withValidListener_callbackClick() {
...
+        verify(listener).onPubnativeNetworkFeedBannerLoadFail(eq(request), eq(PubnativeException.FEED_BANNER_PARAMETERS_INVALID));
+    }
+
+    @Test
+    public void load_withNullContextNullListener_Pass() {
+
+        PubnativeNetworkFeedBanner request = spy(PubnativeNetworkFeedBanner.class);
+        request.mHandler = new Handler();
+
+        request.load(null, TEST_APP_TOKEN, TEST_PLACEMENT_ID_INVALID);
```
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java]: This shouldn't pass since you're giving a null context, so this test should be invoking fail callback

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkInterstitialTest.java [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+        verify(listener).onPubnativeNetworkInterstitialLoadFail(eq(request), eq(PubnativeException.INTERSTITIAL_PARAMETERS_INVALID));
+    }
+
+    @Test
+    public void load_withNullContextNullListener_Pass() {
+
+        PubnativeNetworkInterstitial request = spy(PubnativeNetworkInterstitial.class);
+        request.mHandler = new Handler();
+
+        request.load(null, TEST_APP_TOKEN, TEST_PLACEMENT_ID_INVALID);
```
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkInterstitialTest.java]: This should be invoking fail

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkInterstitialTest.java [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+    @Test
+    public void load_withInvalidParams_callBackLoadFail() {
+
+        PubnativeNetworkInterstitial request = spy(PubnativeNetworkInterstitial.class);
+        PubnativeNetworkInterstitial.Listener listener = spy(PubnativeNetworkInterstitial.Listener.class);
+        request.mHandler = new Handler();
+        request.mListener = listener;
+
+        request.load(RuntimeEnvironment.application.getApplicationContext(), "", "");
```
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkInterstitialTest.java]: This is 2 different tests and should be invoking fail

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java (hunk 6 above)
Comment (anshumansworld) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java]: here listener is also null

File: mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkFeedBannerAdapter.java [hunk 9]
```diff
@@ -203,4 +210,12 @@ protected void invokeClick() {
...
         }
     }
+
+    protected void invokeHide() {
+
+        Log.v(TAG, "invokeHide");
+        if (mAdListener != null) {
+            mAdListener.onAdapterHide(this);
+        }
+    }
```
Comment (cerberillo) [inline mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkFeedBannerAdapter.java]: why are we adding this listener? I don't think it's necessary

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java [hunk 10]
```diff
@@ -150,4 +153,122 @@ public void invokeClick_withValidListener_callbackClick() {
...
+
+        PubnativeNetworkFeedBanner request = spy(PubnativeNetworkFeedBanner.class);
+        PubnativeNetworkFeedBanner.Listener listener = spy(PubnativeNetworkFeedBanner.Listener.class);
+        request.mHandler = new Handler();
+        request.mListener = listener;
+
+        request.load(null, null, null);
+
+        verify(listener).onPubnativeNetworkFeedBannerLoadFail(eq(request), eq(PubnativeException.FEED_BANNER_PARAMETERS_INVALID));
+    }
```
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java]: This test is null considering that you're testing individuals later

**Comment by cerberillo** [comment]
apart from this latest comments, all good to merge

File: mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkFeedBannerAdapter.java (hunk 9 above)
Comment (anshumansworld) [inline mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkFeedBannerAdapter.java]: just to make every standard unit same. This is already added in every type of standard unit (banner, interstitial)

File: mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java (hunk 10 above)
Comment (anshumansworld) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java]: Yes, this is not necessary but I wanted to put one test that is also easily readable. We can remove it as it is not having significant impact. Let me know if you want me to remove this one.

Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/request/PubnativeNetworkFeedBannerTest.java]: yeah, remove this one

File: mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkFeedBannerAdapter.java (hunk 9 above)
Comment (cerberillo) [inline mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkFeedBannerAdapter.java]: ok, let's leave it
